### PR TITLE
[EPPETA-55] Education Preparation Evaluation Tracking Application

### PR DESCRIPTION
**Pull request by jleiva-gap**
- Add optional parameter to EvaluationForm to send the id.
- Update EvaluationForm to load data from an existing evaluation.
- Add alerts and toast to show messages and confirm actions.
- Update page redirects to use navigate.
- Update  GetAllPerformanceEvaluationRatings to include foreign key for Status when the result is returned.
- Add security validation to see the evaluation details (a mentor can only see the evaluations they created, and the supervisor can see other users' evaluations)

### CODE REVIEW

File: src/reactapp/src/pages/EvaluationForm.js [hunk 1]
```diff
@@ -4,119 +4,255 @@
...
+  const [userHasAccessToEvaluation, setUserHasAccessToEvaluation] = useState(true);
+  const [evaluationLoaded, setEvaluationLoaded] = useState(true);
+  const [alertMessageText, setAlertMessageText] = useState(true);
+  const { id } = useParams();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const toast = useToast();  
+  const borderColor = useColorModeValue("gray.200", "gray.600"); 
+  // This will be replaced with data from the ODS/API once EPPETA-38 is complete
+  const ratingLevels = [
```
Comment (stevenarnold-ed-fi) [inline src/reactapp/src/pages/EvaluationForm.js]: @jleiva-gap EPPETA-38 has been completed and merged into main. We're no longer using this hard-coded rating levels object. I noticed the that https://api.ed-fi.org/v6.1/ was missing the rating levels on PerformanceEvaluation again so I've re-added them. Please update this to pull rating level information from the evaluation like in the main branch

Comment (jleiva-gap) [reply src/reactapp/src/pages/EvaluationForm.js]: I updated the PR to remove those hard-coded levels.

File: src/reactapp/src/pages/EvaluationForm.js [hunk 2]
```diff
@@ -4,74 +4,181 @@
...
+      setCurrentEvaluator({ "evaluatorId": getLoggedInUserId(), "evaluatorName": getLoggedInUserName() });
+      loadDataForNewEvaluation();
+    }
+    setIsEvaluationLoaded(true);
+  }, [id]);
+
+  
   const processRatingLevelOptions = async (ratingLevels) => {
-    const processedRatingLevels = [{ "label": "N/A - Not Assessed", "value": -1 }, ...ratingLevels.map((level) => {
+    const processedRatingLevels = [{ "label": "N/A - Not Applicable", "value": -1 }, ...ratingLevels.map((level) => {
```
Comment (stevenarnold-ed-fi) [inline src/reactapp/src/pages/EvaluationForm.js]: @jleiva-gap Please keep the previous terminology of 'Not Assessed'

Comment (jleiva-gap) [reply src/reactapp/src/pages/EvaluationForm.js]: Done

**Review by stevenarnold-ed-fi — CHANGES_REQUESTED** [review]
The code overall looks fine. In testing there is an issue. 
1. Perform an evaluation as a mentor teacher
2. Log in as a supervisor 
3. View and Save the evaluation as supervisor (does not matter if you make changes)
4. Check the database, there are now extra EvaluationElementRatingResults for the evaluation.

This doesn't seem to be an issue with EvaluationRatings or EvaluationObjectiveRatings, just the elementRatingResults

**Comment by DavidJGapCR** [comment]
> The code overall looks fine. In testing there is an issue.
> 
> 1. Perform an evaluation as a mentor teacher
> 2. Log in as a supervisor
> 3. View and Save the evaluation as supervisor (does not matter if you make changes)
> 4. Check the database, there are now extra EvaluationElementRatingResults for the evaluation.
> 
> This doesn't seem to be an issue with EvaluationRatings or EvaluationObjectiveRatings, just the elementRatingResults

I'll take a look at this.

**Review by stevenarnold-ed-fi — COMMENTED** [review]
Due to holiday schedules I'm going to approve this PR with know issues (tracked in EPPETA-68) to be resolved later

**Review by stevenarnold-ed-fi — APPROVED** [review]

**Review by stevenarnold-ed-fi — APPROVED** [review]

**Pull request merged**